services/supervisor/src/simple/mod.rs

```diff
@@ -14,33 +14,31 @@
 
 //! A simple supervisor service that does actions without any confirmations.
 
-pub use proto::ConfigPropose;
-pub use schema::Schema;
-
 use exonum::{
     blockchain::{self, InstanceCollection},
     crypto::Hash,
     helpers::ValidateInput,
     merkledb::Snapshot,
     runtime::{
-        rust::{interfaces::ConfigureCall, BeforeCommitContext, Service, TransactionContext},
-        Caller, ConfigChange, DispatcherError, ExecutionError, InstanceDescriptor, InstanceId,
+        rust::{CallContext, Service},
+        Caller, DispatcherError, ExecutionError, InstanceDescriptor, SUPERVISOR_INSTANCE_ID,
     },
 };
 use exonum_derive::{exonum_service, IntoExecutionError, ServiceFactory};
 
-mod proto;
+use crate::{update_configs, ConfigChange, ConfigPropose, ConfigureCall};
+
 mod schema;
-#[cfg(test)]
-pub mod tests;
+pub use self::schema::Schema;
 
+/// Simple supervisor for testing purposes. **This supervisor is not fit for real-world use cases.**
 #[derive(Debug, ServiceFactory)]
 #[exonum(
-    proto_sources = "proto::schema",
-    artifact_name = "exonum-testkit-supervisor",
-    implements("Transactions")
+    proto_sources = "crate::proto",
+    artifact_name = "simple-supervisor",
+    implements("SupervisorInterface")
 )]
-pub struct SimpleSupervisor;
+pub struct Supervisor;
```

Comment:
Do we actually need this to be a part of `supervisor`? I thought it only exists for test purposes.

Reply:
I've moved it from testkit as to not create a corresponding dependency (I think it's not cyclical, since testkit is a *dev* dependency of the supervisor, but I didn't check). Would you say that if I implemented more "real-world" auth (e.g., by one of validators), `SimpleSupervisor` would make more sense outside of tests?

Reply:
Yeah, I think so.